src/html5/parser/document.rs

```diff
@@ -459,6 +482,39 @@ impl Document {
     }
 }
 
+// Walk the document tree with the given visitor
+pub fn visit(doc: &DocumentHandle, visitor: &mut Box<dyn Visitor<Node>>) {
+    let binding = doc.get();
+    let root = binding.get_root();
+    internal_visit(doc, root, visitor);
+}
+
+fn internal_visit(doc: &DocumentHandle, node: &Node, visitor: &mut Box<dyn Visitor<Node>>) {
```

Comment:
You can get rid of the dyn-Box by using impl:

```suggestion
pub fn visit(doc: &DocumentHandle, visitor: &mut impl Visitor<Node>) {
    let binding = doc.get();
    let root = binding.get_root();
    internal_visit(doc, root, visitor);
}

fn internal_visit(doc: &DocumentHandle, node: &Node, visitor: &mut impl Visitor<Node>) {
```